tests/DesugarTest.elm

```diff
@@ -119,6 +121,32 @@ desugarTest =
                         , insideModule = "A"
                         }
             }
+        , test "desugar duplicate record field" <|
+            \_ ->
+                let
+                    x : Frontend.LocatedExpr
+                    x =
+                        frontendLambda "x" "x"
+
+                    field : Binding Frontend.LocatedExpr
+                    field =
+                        { name = "a", body = x }
+
+                    fields : List (Binding Frontend.LocatedExpr)
+                    fields =
+                        [ field, field ]
+                in
+                located (Frontend.Record fields)
+                    |> Desugar.desugarExpr Dict.empty (moduleFromName "A")
+                    |> mapUnwrap
+                    |> Expect.equal
+                        (Err <|
+                            CompilerError.DuplicateRecordField
+                                { name = "a"
+                                , firstOccurrence = x |> Located.replaceWith ()
+                                , secondOccurrence = x |> Located.replaceWith ()
```

Comment:
If we want to have the two occurence fields in the error even though we ignore them for now, perhaps we should check that they contain expected data? (You'd probably have to *not* use `located` helper for that particular test then.)

(Removing the occurence might be easier for now, and I wouldn't be against that :slightly_smiling_face: )